File: sdk/nodejs/src/resources/service_status.js
"use strict";

const { BaseResource } = require("./base");

class ServiceStatusResource extends BaseResource {
  async list(params = {}) {
    return this._http.get("/api/statuses");
  }

  async update(statusId, data) {
    return this._http.put("/api/statuses/{id}", data);
  }

}

module.exports = { ServiceStatusResource };

File: sdk/java/src/main/java/vn/hitechcloud/sdk/resources/ServiceStatus.java
package vn.hitechcloud.sdk.resources;

import java.util.Map;
import vn.hitechcloud.sdk.HttpClient;

public class ServiceStatus extends BaseResource {

    public ServiceStatus(HttpClient http) {
        super(http);
    }

    /** List statuses */
    public Map<String, Object> statuses() throws Exception {
        return http.get("/api/statuses");
    }

    /** Update status */
    public Map<String, Object> updateStatus(int id, Map<String, Object> data) throws Exception {
        return http.put(String.format("/api/statuses/%id", id), data);
    }

}
